src/coreclr/src/CMakeLists.txt

```diff
@@ -60,6 +60,14 @@ if(CLR_CMAKE_HOST_UNIX)
   add_subdirectory(nativeresources)
 endif(CLR_CMAKE_HOST_UNIX)
 
+# Workaround for errors on GCC
+if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+    if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 8.0)
+        add_compile_options(-Wno-error=stringop-overflow=)
+    endif()
+    add_compile_options(-Wno-error=tautological-compare)
```

Comment:
This workaround is not required. CI has v7 working just fine without it.

Reply:
v7 works fine, but failed on v10.2. (I'm not sure which version the problem started)

```
  /home/heiher/git/dotnet/runtime/src/coreclr/src/vm/methodtable.cpp: In instantiation of ‘struct MethodTable::MultipurposeSlotOffset<0>’:
  /home/heiher/git/dotnet/runtime/src/coreclr/src/vm/methodtable.cpp:349:5:
   required from here
  /home/heiher/git/dotnet/runtime/src/coreclr/src/vm/methodtable.cpp:327:33:
 error: bitwise comparison always evaluates to false [-Werror=tautological-compare]
    327 |     enum { index = (((mask & 3) == 2) && (raw == 1)) ? 0 : raw };
        |                     ~~~~~~~~~~~~^~~~~
```

Reply:
This workaround is applying to all versions and the line above is applying to 8.0. We have CI logs for v7 and v8.4 from < 24 hrs. old master which shows that it is not required for those versions of gcc.

In the past we have only added conditions based on the versions against which errors were found. So in this case, you can apply suppression to version 10 against which you have observed these errors.

Reply:
OK, thank you.